cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -477,6 +491,87 @@ class ArrayCountSorter {
   }
 };
 
+using ::arrow::internal::Bitmap;
+
+template <>
+class ArrayCountSorter<BooleanType> {
+ public:
+  ArrayCountSorter() = default;
+
+  // Returns where null starts.
+  uint64_t* Sort(uint64_t* indices_begin, uint64_t* indices_end,
+                 const BooleanArray& values, int64_t offset,
+                 const ArraySortOptions& options) {
+    // 32bit counter performs much better than 64bit one
```

Comment:
This sounds rather weird. Are you sure about this?

Reply:
well, I didn't test this. Just following the `ArrayCountSorter` impl. 
https://github.com/apache/arrow/blob/36d7a562f08d77dbf5ed699d486badcc9403f619/cpp/src/arrow/compute/kernels/vector_sort.cc#L438

Reply:
This comment was added by me very early when implementing counting sort approach.
I think it's possibly due to 32bit counter array is smaller and have more chance to stay in L1 cache.

Reply:
Unless it's benchmarked here as well, maybe let's remove the comment to avoid being confusing.

Reply:
This comment is there in the primitive type impl as well

Reply:
That one was benchmarked!

More seriously…maybe at least edit them to reflect that it was done due to a benchmark. Though I worry about the comment effectively bitrotting.